### PR TITLE
[release/5.3] Issue #794 Degrade when memory pools not identified

### PR DESCRIPTION
**Pull request by ljacomet**
Also update the max direct memory value used on command line, IBM VM no
longer parses 2^63-1 properly

### CODE REVIEW

File: terracotta-kit/src/assemble/server/bin/start-tc-server.bat [hunk 1]
```diff
@@ -48,11 +48,11 @@ if not defined JAVA_HOME (
...
-			"\bin\java -server -XX:MaxDirectMemorySize=9223372036854775807" ^
-			"\bin\java -d64 -client  -XX:MaxDirectMemorySize=9223372036854775807" ^
-			"\bin\java -client -XX:MaxDirectMemorySize=9223372036854775807" ^
-			"\bin\java -XX:MaxDirectMemorySize=9223372036854775807") ^
+for %%C in ("\bin\java -d64 -server -XX:MaxDirectMemorySize=1048576g" ^
+			"\bin\java -server -XX:MaxDirectMemorySize=1048576g" ^
+			"\bin\java -d64 -client  -XX:MaxDirectMemorySize=1048576g" ^
+			"\bin\java -client -XX:MaxDirectMemorySize=1048576g" ^
+			"\bin\java -XX:MaxDirectMemorySize=1048576g") ^
 do (
```
Comment (myronkscott) [inline terracotta-kit/src/assemble/server/bin/start-tc-server.bat]: Maybe we try both values in the for loop?  I'm worried about other platforms not handling the new style.

Comment (ljacomet) [reply terracotta-kit/src/assemble/server/bin/start-tc-server.bat]: This has been tested with an Oracle JVM and the IBM one. Which other are you thinking of?

Comment (lorban) [reply terracotta-kit/src/assemble/server/bin/start-tc-server.bat]: `Long.MAX_VALUE - 1` just proved to be problematic so I definitely wouldn't rely on it anymore. Having spent a few minutes in the Hotspot initialization code that slices up the native heap, it already is almost a miracle that this value ever worked at all.

This format is standard for `-Xmx` so I'm tempted to believe the same code will always be used to parse the `XX` memory size flags as well. We could convert the 1 PB @ljacomet used here from gigabytes down to bytes but that sounds to me like splitting hair.

Comment (lorban) [reply terracotta-kit/src/assemble/server/bin/start-tc-server.bat]: For future reference: http://mail.openjdk.java.net/pipermail/hotspot-runtime-dev/2017-April/023124.html

**Pull request merged**